### ISSUE TITLE
`let_unit_value` mangles field init shorthand syntax

### ISSUE THREAD

**Issue by ada4a** [rust-lang/rust-clippy#15789]
### Summary

If the name of the local with a unit value is the same as one of the fields of a struct, and the local is used in a constructor of that struct, then the whole field expression gets replaced by `()`.

This is a minimized reproducer of https://docs.rs/bitflags/2.6.0/src/bitflags/parser.rs.html#283

### Reproducer

I tried this code:

```rust
#![allow(unused)]
#![warn(clippy::let_unit_value)]

struct Foo {
    value: (),
}

fn foo() -> Foo {
    let value = println!();

    Foo { value }
}
```

I expected to see this happen:
```
warning: this let-binding has unit value
 --> src/main.rs:9:5
  |
9 |     let value = println!();
  |     ^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#let_unit_value
note: the lint level is defined here
 --> src/main.rs:2:9
  |
2 | #![warn(clippy::let_unit_value)]
  |         ^^^^^^^^^^^^^^^^^^^^^^
help: omit the `let` binding and replace variable usages with `()`
  |
9 ~     println!();
10|
11~     Foo { value: () }
  |
```

Instead, this happened:
```
help: omit the `let` binding and replace variable usages with `()`
  |
9 ~     println!();
10|
11~     Foo { () }
  |
```

### Version

```text
rustc 1.90.0 (1159e78c4 2025-09-14)
binary: rustc
commit-hash: 1159e78c4747b02ef996e55082b704c09b970588
commit-date: 2025-09-14
host: x86_64-unknown-linux-gnu
release: 1.90.0
LLVM version: 20.1.8
```

### Additional Labels

@rustbot I-suggestion-causes-error
